src/convertToObject.js

```diff
@@ -5,8 +5,48 @@
  *
  * @return {object}
  */
-function convertToObject(sourceString) {
-  // write your code here
-}
+const convertToObject = function (sourceString) {
+  const styleObject = {};
+
+  const cleanSourceString = sourceString
+    .replace(/[\t\r]+/g, '')
+    .replace(/\s*;\s*/g, ';')
+    .replace(/\s*:\s*/g, ':')
+    .replace(/;\n/g, ';');
+
+  const declarationsArray = cleanSourceString.trim().split(';').filter(Boolean);
+
+  let buffer = '';
+  let insideMultilineValue = false;
```

Comment:
[CODE PRACTICE] - The use of a buffer and a flag to handle multiline values is unnecessary for the given task, as handling cases with multiple colons within the style string's property values is not required. Consider simplifying your code by removing these constructs.